### PR TITLE
bugfix: Fix StacktraceAnalyser in case of empty package

### PR DESCRIPTION
**Pull request by tgodzik**
Extracted from https://github.com/scalameta/metals/pull/5197

### CODE REVIEW

**Review by kasiaMarek — COMMENTED** [review]
You've added the empty package in case of top-level method definition. Don't you need to do the same for other cases when there is no package?

**Comment by tgodzik** [comment]
> You've added the empty package in case of top-level method definition. Don't you need to do the same for other cases when there is no package?

You're right, made it more general and added a test case

File: metals/src/main/scala/scala/meta/internal/metals/StacktraceAnalyzer.scala [hunk 1]
```diff
@@ -210,10 +210,13 @@ object StacktraceAnalyzer {
     /* Symbol containing `$package$` is a toplevel method and we only need to
      * find any method contained in the same file even if overloaded
      */
-    if (symbol.contains("$package$")) {
+    val symbolToFind = if (symbol.contains("$package$")) {
       symbolIn.split("\\$package\\$") match {
         case Array(filePath, symbol) =>
-          val re = filePath.replace('.', '/') + "$package" + symbol
```
Comment (jkciesluk) [inline metals/src/main/scala/scala/meta/internal/metals/StacktraceAnalyzer.scala:216]: Do we need this change?

**Review by jkciesluk — APPROVED** [review]
lgtm

File: metals/src/main/scala/scala/meta/internal/metals/StacktraceAnalyzer.scala (hunk 1 above)
Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/metals/StacktraceAnalyzer.scala:216]: Good catch, we don't!

**Pull request merged**